src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,152 @@
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == threshold) {
+            resize();
+        }
 
+        if (key == null) {
+            putNullValue(value);
+            return;
+        }
+        int index = hash(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] != null) {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (currentNode.key.equals(key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = newNode;
+                    size++;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+        } else {
+            table[index] = newNode;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            getNullValue();
+        }
+        int index = hash(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (currentNode.key.equals(key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private V putNullValue(V value) {
+        if (table[0] != null) {
+            Node<K, V> currentNode = table[0];
+            while (currentNode != null) {
+                if (currentNode.next == null) {
+                    currentNode.value = value;
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next;
+            }
+        } else {
+            table[0] = new Node<>(null, value, null);
+            size++;
+        }
+        return null;
+    }
+
+    private V getNullValue() {
+        Node<K, V> currentNode = table[0];
+        while (currentNode != null) {
+            if (currentNode.key == null) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode() % table.length);
+    }
+
+    private void resize() {
+        int newCapacity = table.length << 1;
+        Node<K,V>[] oldTable = table;
+        threshold = (int) (newCapacity * DEFAULT_LOAD_FACTOR);
+        table = new Node[newCapacity];
+
+        for (Node<K,V> node : oldTable) {
+            while (node != null) {
+                Node<K,V> nextNode = node.next;
+                node.next = null;
+                int index = hash(node.key);
+                node.next = table[index];
+                table[index] = node;
+                node = nextNode;
```

Comment:
Consider using put method when resizing

Reply:
if i simplify it to
```
for (Node<K, V> node : oldTable) {
            while (node != null) {
                put(node.key, node.value);
                node = node.next;
            }
        }
```
i will get three problems

@Test
    public void getSizeWithCollisionAtFirstPosition() {
    
     public void getSizeWithCollisionAtZeroPosition() {
     
     public void checkTheHashMapIncrease() {